recipes/folly/all/conanfile.py

```diff
@@ -39,49 +27,73 @@ def _source_subfolder(self):
-            raise ConanInvalidConfiguration("Folly could not be built by Visual Studio < 14")
-        elif self.settings.os == "Windows" and \
-            self.settings.arch != "x86_64":
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++14 support. The current compiler {} {} does not support it.".format(
+                    self.name, self.settings.compiler, self.settings.compiler.version))
+
+        if self.settings.os == "Windows" and self.settings.arch != "x86_64":
             raise ConanInvalidConfiguration("Folly requires a 64bit target architecture")
-        elif self.settings.os == "Windows" and \
-            self.settings.compiler == "Visual Studio" and \
-            "MT" in self.settings.compiler.runtime:
+        elif self.settings.os == "Windows" and self.settings.compiler == "Visual Studio" and \
+                "MT" in self.settings.compiler.runtime:
             raise ConanInvalidConfiguration("Folly could not be build with runtime MT")
-        elif self.settings.os == "Linux" and \
-            self.settings.compiler == "clang" and \
-            compiler_version < "6.0":
-            raise ConanInvalidConfiguration("Folly could not be built by Clang < 6.0")
-        elif self.settings.os == "Linux" and \
-            self.settings.compiler == "gcc" and \
-            compiler_version < "5":
-            raise ConanInvalidConfiguration("Folly could not be built by GCC < 5")
-        elif self.settings.os == "Macos" and \
-            self.settings.compiler == "apple-clang" and \
-            compiler_version < "8.0":
-            raise ConanInvalidConfiguration("Folly could not be built by apple-clang < 8.0")
         elif self.settings.os == "Macos" and self.options.shared:
             raise ConanInvalidConfiguration("Folly could not be built by apple-clang as shared library")
-        elif self.settings.os == "Windows" and \
-             self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built on Windows as shared library")
+        elif self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Folly could not be built on Windows as a shared library")
+        elif Version(self.version) >= "2020.08.10.00" and self.settings.compiler == "Visual Studio" and \
+                not self.options.shared:
+            raise ConanInvalidConfiguration("Folly could not be built on Windows as a static library")
```

Comment:
Folly can't be built as static if Windows or Visual Studio only?

Visual Studio is completly dropped (shared and static) in 2020.08.10.00?

Reply:
Well, I've tested, and MinGW is just not supported.
So Windows support is completly removed in folly recipe starting from 2020.08.10.00 :s

Reply:
Yes I had I nightmare trying to sort of the issues. I suspect there'll be another version in the near future that will revive it...

Reply:
https://github.com/conan-io/conan-center-index/pull/2535#issuecomment-686847741, I did not re-open the logs but i had looked into locally for a bit and there was no obvious answer.

Reply:
I wonder what is the goal to have this package if we cannot use it on Windows, any solutions ?

Reply:
If you look at the logs from #3177 

```
C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(417): message : while compiling class template member function 'void folly::SharedMutexImpl<false,void,std::atomic,false,false>::lock_shared(folly::SharedMutexImpl<false,void,std::atomic,false,false>::Token &)' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1362): message : see reference to function template instantiation 'void folly::SharedMutexImpl<false,void,std::atomic,false,false>::lock_shared(folly::SharedMutexImpl<false,void,std::atomic,false,false>::Token &)' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1571): message : see reference to class template instantiation 'folly::SharedMutexImpl<false,void,std::atomic,false,false>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/lang/Align.h(105): message : see reference to class template instantiation 'folly::detail::max_align_t_<long double,double,float,__int64,long,int,short,bool,char,char16_t,char32_t,wchar_t,void *,std::max_align_t>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\chrono(632): message : see reference to class template instantiation 'std::chrono::duration<double,std::ratio<1,1>>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\chrono(178): message : see reference to class template instantiation 'std::chrono::duration<__int64,std::nano>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\chrono(610): message : see reference to class template instantiation 'std::chrono::time_point<std::chrono::steady_clock,std::chrono::nanoseconds>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1673,1): error C2783: 'folly::atomic_ref<T> folly::make_atomic_ref_t::operator ()(T &) const': could not deduce template argument for '__formal' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/synchronization/AtomicRef.h(142): message : see declaration of 'folly::make_atomic_ref_t::operator ()' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1698,13): error C2672: 'operator __surrogate_func': no matching overloaded function found [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1698,1): error C2783: 'folly::atomic_ref<T> folly::make_atomic_ref_t::operator ()(T &) const': could not deduce template argument for '__formal' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/synchronization/AtomicRef.h(142): message : see declaration of 'folly::make_atomic_ref_t::operator ()' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1750,7): error C2672: 'operator __surrogate_func': no matching overloaded function found [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1750,1): error C2783: 'folly::atomic_ref<T> folly::make_atomic_ref_t::operator ()(T &) const': could not deduce template argument for '__formal' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/synchronization/AtomicRef.h(142): message : see declaration of 'folly::make_atomic_ref_t::operator ()' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]
```

It's really bad folly code that prevents it from compiling in windows.... The mistake on my part was not putting a better exception message... that version does not compile on windows

It's unfortunate that it takes so much effort to add a new version and I only noticed the one I started on was bad in windows.